src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -57,11 +69,179 @@ public function getStatesAction(Request $request)
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_states_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/add.html.twig', [
+            'enableSidebar' => true,
+            'stateForm' => $stateForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * Handles edit form rendering and submission
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_states_index"
+     * )
+     *
+     * @param int $stateId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(int $stateId, Request $request): Response
+    {
+        try {
+            /** @var EditableState $editableState */
+            $editableState = $this->getQueryBus()->handle(new GetStateForEditing((int) $stateId));
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        }
+
+        $stateForm = null;
+
+        try {
+            $stateFormBuilder = $this->get(
+                'prestashop.core.form.identifiable_object.builder.state_form_builder'
+            );
+
+            $stateFormHandler = $this->get(
+                'prestashop.core.form.identifiable_object.handler.state_form_handler'
+            );
+
+            $stateForm = $stateFormBuilder->getFormFor((int) $stateId);
+            $stateForm->handleRequest($request);
+            $result = $stateFormHandler->handleFor((int) $stateId, $stateForm);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_states_index');
+            }
+        } catch (StateNotFoundException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('Edit: %value%', 'Admin.Actions', ['%value%' => $editableState->getName()]),
+            'stateForm' => $stateForm->createView(),
```

Comment:
I think static php analyzer would tell you that $stateForm might not have been defined.
you could check it just to make sure, and redirect to index page if it isn't.

Reply:
what do you mean by static page analyzer?

Reply:
phpstan for example.

Reply:
well, $stateForm is inside try catch block; would be nice if phpstan also could adjust to that. I would prefer to use try catch instead of ifs..